### ISSUE TITLE
can't save Model objects with xarray properties

### ISSUE THREAD

**Issue by barkls** [manoharan-lab/holopy#256]
When saving a model by itself or as part of a results object, pack_attrs or equivalent is never run, so we have yaml.dump of xarray objects happening. These can't be loaded back up.

Possible solutions:
1) rewrite pack_attrs to work on model, etc.
2) write yaml constructor and representer classes for xarray objects

**Comment by barkls** [comment]
quick fix: disallow passing in xarrays for alpha, noise_sd, etc.

**Comment by briandleahy** [comment]
Hey @barkls can I get close to a MWE? I'm not sure if the problem is something like this:
```python
import xarray as xr
from holopy.inference import AlphaModel, prior
from holopy.scattering.scatterer import Sphere

scatterer = Sphere(n=1.59, r=1.0)

model = AlphaModel(scatterer, noise_sd=xr.DataArray(0.5))
```
or this
```python
model = AlphaModel(scatterer, noise_sd=prior.Uniform(xr.DataArray(0.0), xr.DataArray(0.5)))
```

Thanks!

**Comment by barkls** [comment]
I think your first suggestion is right. Can you save that object with
hp.save? Does hp.load on the file then fail?

I can work on putting on an example but probably not until Thursday.

Solomon

**Comment by briandleahy** [comment]
An example would be helpful. Thanks!

**Comment by barkls** [comment]
```python
import tempfile
import yaml
import xarray as xr
import holopy as hp
from holopy.core import detector_grid, update_metadata

def generate_model(alpha, polarization=None):
    scatterer = hp.scattering.Sphere(r=0.5, n=1.5, center=(1,1,1))
    return hp.inference.AlphaModel(scatterer, alpha=alpha)

def file_round_trip(model):
    tempdir = tempfile.mkdtemp()
    hp.save(tempdir + 'test.h5', model)
    hp.load(tempdir + 'test.h5')

def yaml_round_trip(model):
    object_string = yaml.dump(model)
    yaml.load(object_string, Loader=hp.core.holopy_object.FullLoader)

alpha_xarray = xr.DataArray([1,0.5],
                            dims=['illumination'],
                            coords={'illumination':['red', 'green']})
alpha_dict = {'red':1, 'green':0.5}

# No problem if alpha is a dictionary
file_round_trip(generate_model(alpha_dict))
yaml_round_trip(generate_model(alpha_dict))

# Cannot save and reload a file if alpha is a xr.DataArray
file_round_trip(generate_model(alpha_xarray))

# The problem is because yaml can't parse the xarray data
yaml_round_trip(generate_model(alpha_xarray))

# This applies also to noise_sd, medium_wavelen, illum_polarization etc.
# Note that illum_polarization is always a vector if coming from an image
dummy_detector = update_metadata(detector_grid(4, 1), illum_polarization=(0,1))
model = generate_model(alpha_dict)
model.illum_polarization = dummy_detector.illum_polarization
yaml_round_trip(model)

**Comment by barkls** [comment]
@briandleahy introduced a temporary fix in #296 by forbidding creation of Model objects with xarray.Datarrays in them. However the longer-term fix is probably to write yaml representer/constructor classes that work with simple xarrays (assuming we keep using yaml and xarray).
